src/index/index.rs

```diff
@@ -560,32 +554,40 @@ impl Index {
                     ),
                 )
             })?;
-        let memory_arena_in_bytes_per_thread = overall_memory_budget_in_bytes / num_threads;
-        IndexWriter::new(
-            self,
-            num_threads,
-            memory_arena_in_bytes_per_thread,
-            directory_lock,
-            num_merge_threads,
-        )
+
+        IndexWriter::new(self, options, directory_lock)
     }
 
-    /// Creates a multithreaded writer with 4 merge threads.
+    /// Open a new index writer. Attempts to acquire a lockfile.
+    ///
+    /// The lockfile should be deleted on drop, but it is possible
+    /// that due to a panic or other error, a stale lockfile will be
+    /// left in the index directory. If you are sure that no other
+    /// `IndexWriter` on the system is accessing the index directory,
+    /// it is safe to manually delete the lockfile.
+    ///
+    /// - `num_threads` defines the number of indexing workers that should work at the same time.
+    ///
+    /// - `overall_memory_budget_in_bytes` sets the amount of memory allocated for all indexing
+    ///   thread.
+    ///
+    /// Each thread will receive a budget of `overall_memory_budget_in_bytes / num_threads`.
     ///
     /// # Errors
-    /// If the lockfile already exists, returns `Error::FileAlreadyExists`.
+    /// If the lockfile already exists, returns `Error::DirectoryLockBusy` or an `Error::IoError`.
     /// If the memory arena per thread is too small or too big, returns
     /// `TantivyError::InvalidArgument`
     pub fn writer_with_num_threads<D: Document>(
         &self,
         num_threads: usize,
         overall_memory_budget_in_bytes: usize,
     ) -> crate::Result<IndexWriter<D>> {
-        self.writer_with_num_threads_and_num_merge_threads(
-            num_threads,
-            overall_memory_budget_in_bytes,
-            DEFAULT_NUM_MERGE_THREADS,
-        )
+        let memory_arena_in_bytes_per_thread = overall_memory_budget_in_bytes / num_threads;
```

Comment:
can we enforce one way or another that num_threads is > 0.

(Just an assert with an error message is ok...)

Reply:
Yes will add

Reply:
last remaining point here.